File: src/main/java/no/novari/flyt/instance/gateway/model/vigo/Dokument.java
package no.novari.flyt.instance.gateway.model.vigo;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;
import no.novari.flyt.instance.gateway.validation.constraints.ValidBase64;

@Getter
@Jacksonized
@EqualsAndHashCode
@Builder
public class Dokument {

    private String tittel;
    private String dato;
    private String filnavn;

    @ValidBase64
    private String fil;
    private String format;
}

File: src/main/kotlin/no/fintlabs/instance/gateway/model/vigo/Dokument.kt
package no.fintlabs.instance.gateway.model.vigo

import com.fasterxml.jackson.annotation.JsonInclude
import no.fintlabs.gateway.instance.validation.constraints.ValidBase64

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Dokument(
    val tittel: String? = null,
    val dato: String? = null,
    val filnavn: String? = null,
    @field:ValidBase64
    val fil: String? = null,
    val format: String? = null
)
